### PR TITLE
[State] Add abstract State and PositionState

### PR DESCRIPTION
**Pull request by Herklos**

### CODE REVIEW

File: octobot_trading/personal_data/positions/position_state.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+#  License along with this library.
+
+import octobot_trading.constants
+import octobot_trading.enums as enums
+import octobot_trading.errors
+import octobot_trading.exchange_channel as exchange_channel
+import octobot_trading.personal_data.state as state_class
+
+
+class PositionState(state_class.State):
```
Comment (GuillaumeDSM) [inline octobot_trading/personal_data/positions/position_state.py:24]: 👍

File: octobot_trading/exchanges/types/future_exchange.py [hunk 2]
```diff
@@ -195,7 +195,10 @@ def parse_position_status(self, status):
         :param status: the position raw status
         :return: the uniformized position status
         """
-        raise NotImplementedError("parse_position_status is not implemented")
+        try:
+            return octobot_trading.enums.PositionStatus(status)
+        except KeyError:
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/types/future_exchange.py:200]: I think it is a ValueError when an enum is not found

File: octobot_trading/personal_data/positions/channel/positions_updater.pxd [hunk 3]
```diff
@@ -16,6 +16,13 @@
...
 cimport octobot_trading.personal_data.positions.channel.positions as positions_channel
 
+cimport octobot_commons.async_job as async_job
 
 cdef class PositionsUpdater(positions_channel.PositionsProducer):
     cdef public bint should_use_open_position_per_symbol
+
+    cdef async_job.AsyncJob open_positions_job
+    cdef async_job.AsyncJob closed_positions_job
+    cdef async_job.AsyncJob position_update_job
```
Comment (GuillaumeDSM) [inline octobot_trading/personal_data/positions/channel/positions_updater.pxd:26]: 👍

File: octobot_trading/personal_data/positions/channel/positions_updater.py [hunk 4]
```diff
@@ -16,69 +16,185 @@
...
 
-            except errors.NotSupported:
-                self.logger.warning(f"{self.channel.exchange_manager.exchange_name} is not supporting updates")
-                await self.stop()
-            except Exception as e:
-                self.logger.exception(e, True, f"Fail to update positions : {e}")
+        await self.initialize()
+        await asyncio.sleep(self.POSITIONS_STARTING_REFRESH_TIME)
+        await self.open_positions_job.run()
+        # await self.closed_orders_job.run()
```
Comment (GuillaumeDSM) [inline octobot_trading/personal_data/positions/channel/positions_updater.py:84]: should we remove this line ?

File: octobot_trading/personal_data/positions/position_state.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    def __init__(self, position, is_from_exchange_data):
+        super().__init__(is_from_exchange_data)
+
+        # ensure position has not been cleared
+        if position.is_cleared():
+            raise octobot_trading.errors.InvalidOrderState(f"Position has already been cleared. Position: {position}")
```
Comment (GuillaumeDSM) [inline octobot_trading/personal_data/positions/position_state.py:30]: we should add an `__str__` method to `Position`

File: octobot_trading/exchanges/types/future_exchange.py (hunk 2 above)
Comment (Herklos) [reply octobot_trading/exchanges/types/future_exchange.py:200]: Right

**Review by GuillaumeDSM — APPROVED** [review]
👍

**Pull request merged**